### PR TITLE
Add numpy array support for orjson serialization

### LINKED ISSUE #7910: Float64-only pandas Dataframes not supported as flow arguments
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Flows don't support pandas Dataframes if they contain only float64 columns. Once we add any non-float64 (float32/int etc) column to the dataframe, serialization of the parameters works fine. I'm not sure why this makes any difference. 

Adding `orjson.OPT_SERIALIZE_NUMPY` flag to `prefect.orion.utilities.schemas.orjson_dumps_non_str_keys` solves this issue.

### Reproduction

```python3
import pandas as pd
from prefect import flow


@flow(validate_parameters=False)
def transformer(df: pd.DataFrame):
    return df


x = pd.DataFrame({
    "a": [0.],
    # "b": [0]  # uncomment to make it work
})
print(x.dtypes)
transformer(x)
```

### Error

```python3
Traceback (most recent call last):
  File "/Users/michal.augoff/Documents/GitHub/mls-mlops/local_run.py", line 15, in <module>
    transformer(x)
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/prefect/flows.py", line 447, in __call__
    return enter_flow_run_engine_from_flow_call(
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/prefect/engine.py", line 162, in enter_flow_run_engine_from_flow_call
    return anyio.run(begin_run)
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/prefect/client/utilities.py", line 47, in with_injected_client
    return await fn(*args, **kwargs)
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/prefect/engine.py", line 219, in create_then_begin_flow_run
    flow_run = await client.create_flow_run(
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/prefect/client/orion.py", line 449, in create_flow_run
    flow_run_create_json = flow_run_create.dict(json_compatible=True)
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/prefect/orion/utilities/schemas.py", line 268, in dict
    return json.loads(self.json(*args, **kwargs))
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/prefect/orion/utilities/schemas.py", line 238, in json
    return super().json(*args, **kwargs)
  File "pydantic/main.py", line 505, in pydantic.main.BaseModel.json
  File "/Users/michal.augoff/anaconda3/envs/mlops2/lib/python3.8/site-packages/prefect/orion/utilities/schemas.py", line 135, in orjson_dumps_non_str_keys
    return orjson.dumps(v, default=default, option=orjson.OPT_NON_STR_KEYS).decode()
TypeError: Type is not JSON serializable: numpy.float64
```

### Versions

```Text
2.7.1
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Ewande**
Closes #7910 by enabling Numpy array serialization in `orjson.dumps`

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: src/prefect/orion/utilities/schemas.py [hunk 1]
```diff
@@ -126,13 +126,16 @@ def orjson_dumps(v: Any, *, default: Any) -> str:
 
 def orjson_dumps_non_str_keys(v: Any, *, default: Any) -> str:
```
Comment (madkinsz) [inline src/prefect/orion/utilities/schemas.py:127]: We'll probably want to rename this as like... `orjson_dumps_extra_compatible` or something :D (maybe there's a better name than that)

Comment (Ewande) [reply src/prefect/orion/utilities/schemas.py]: good catch, thanks! I also added a test for compatibility with numpy types but this introduces a new requirement in `requirements-dev.txt`, so let me know if you want to maintain it

**Review by madkinsz — APPROVED** [review]
@Ewande no need to keep it up to date with `main`

I'm hesitant to add `numpy` for all developers, we'll look into a way to lower the footprint of that separately though. 

@zangell44 can you please review.

**Comment by Ewande** [comment]
hey team, is there a chance I can get this PR reviewed?

**Review by madkinsz — APPROVED** [review]
@zangell44 pretty please :)

**Review by zangell44 — APPROVED** [review]

**Pull request merged**